torch/nn/_functions/rnn.py

```diff
@@ -8,19 +8,22 @@
 
 
 def RNNReLUCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
-    hy = F.relu(F.linear(input, w_ih, b_ih) + F.linear(hidden, w_hh, b_hh))
+    x_h = input if w_ih is None else F.linear(input, w_ih, b_ih)
+    hy = F.relu(x_h + F.linear(hidden, w_hh, b_hh))
     return hy
 
 
 def RNNTanhCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
-    hy = F.tanh(F.linear(input, w_ih, b_ih) + F.linear(hidden, w_hh, b_hh))
+    x_h = input if w_ih is None else F.linear(input, w_ih, b_ih)
+    hy = F.tanh(x_h + F.linear(hidden, w_hh, b_hh))
     return hy
 
 
 def LSTMCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
     hx, cx = hidden
-    gates = F.linear(input, w_ih, b_ih) + F.linear(hx, w_hh, b_hh)
-    ingate, forgetgate, cellgate, outgate = gates.chunk(4, 1)
+    x_h = input.unsqueeze(1).expand(input.size(0), 4, input.size(1)) if w_ih is None else F.linear(input, w_ih, b_ih)
+    gates = x_h + F.linear(hx, w_hh, b_hh)
+    ingate, forgetgate, cellgate, outgate = [x.squeeze(1) for x in gates.chunk(4, 1)]
```

Comment:
You're relying on the fact that (for the non-skip_input case), dim>1, otherwise it will squeeze this to a 1D tensor.

Better to just match the dimension for skip_input and regular:

`x_h = input.repeat(1, 4) if w_ih is None else F.linear(input, w_ih, b_ih)`

Then you can remove the `squeeze`.